dd-sdk-android-ndk/src/main/cpp/utils/signal-monitor.c

```diff
@@ -19,21 +20,24 @@
 
 #include "../datadog-native-lib.h"
 
-void handle_signal(int signum, siginfo_t *info,
-                   void *user_context);
-
-bool configure_signal_stack(void);
+static const char *LOG_TAG = "DatadogNdkCrashReporter";
 
 stack_t signal_stack;
-/**
- * Global shared context reports
- */
-static JNIEnv *global_env;
+
 /* the signal handler array */
 struct sigaction *global_sigaction;
 
 /* the previous signal handler array */
 struct sigaction *global_previous_sigaction;
+sigset_t signals_mask;
+
+pthread_t datadog_watchdog_thread;
+
+bool handlers_installed = false;
+
+// for testing purposes
+int performed_install_ops = 0;
+int performed_uninstall_ops = 0;
```

Comment:
Not an expert here (Simon can correct me) but if it's for test purposes only, can we maybe guard those declaration between an `#IFDEF` ? That way the release version of the lib won't have those ? That's not that important (just 2 ints here) but if our tests get more complex, it'd be nice to have a way to clean the code for released libs.

Reply:
yeah that's a nice idea...I will guard those ;)